src/test/java/org/kohsuke/github/GitHubTest.java

```diff
@@ -124,6 +124,11 @@ public void testGitHubEnterpriseDoesNotHaveRateLimit() throws IOException {
     @Test
     public void testGitHubIsApiUrlValid() throws IOException {
         GitHub github = GitHub.connectAnonymously();
-        github.checkApiUrlValidity();
+        //GitHub github = GitHub.connectToEnterpriseAnonymously("https://github.mycompany.com/api/v3/");
```

Comment:
@cyrille-leclerc You can use this test to play with the method and different scenarios.